Added a 'no-build' option for docker-compose

RmStorm opened the pull request:
Hi, I have a fairly large docker-compose file and the build command significantly slows down my testing. Especially when writing tests it is quite unnecessary to rebuild the unchanged code in the container every time.

I have extended the plugin with an optional flag. The default behaviour stays identical but calling pytest with --docker-compose-no-build prevents the plugin from building the containers.

RmStorm commented:
Hi, do you think it will be possible to merge this into master and publish it on pypi? Otherwise I will have to look into a different way to package this since I would like to implement automated integration tests on a few projects.

todofixthis commented:
Hi @RmStorm I love this idea and would love to see it included!  I've pinged a couple of folks at Centrality to see about getting this merged 😺

RmStorm commented:
Thank you for the reply, it would be nice if it's published but it's not important for me anymore. I have published a new version with a new name in a private repository to avoid confusion on pypi.
I added tests, testing against python 3.5 and 3.6 and CI/CD, please see [here](https://github.com/RmStorm/pytest-docker-compose/tree/gemfury-master).
I would like to publish those things on the repository here and hook it up to pypi instead of a private gemfury repo. Up to you guys if that's going to work out!

I also created a dynamically scoped container fixture and added error messaging when it's used incorrectly. E.g. trying to spin up multiple conflicting docker-compose networks due to poor scoping.

markf94 commented:
This is amazing! I started using `pytest-docker-compose` this morning, encountered exactly the same issue and then discovered an hour ago that you guys just merged the fix. Most epic coincidence ever. Thanks @RmStorm for this contribution :)
